gammapy/background/reflected.py

```diff
@@ -236,45 +241,40 @@ def plot(self, fig=None, ax=None):
             off_patch = tmp.as_artist(color="blue", alpha=0.6)
             ax.add_patch(off_patch)
 
-            test_pointing = self.center
-            ax.scatter(
-                test_pointing.galactic.l.degree,
-                test_pointing.galactic.b.degree,
-                transform=ax.get_transform("galactic"),
+            xx, yy = self.center.to_pixel(wcs)
+            ax.plot(
+                xx, yy,
                 marker="+",
-                s=300,
-                linewidths=3,
                 color="green",
+                markersize=20,
+                linewidth=5,
             )
 
         return fig, ax
 
-    @staticmethod
-    def _is_inside_exclusion(pixreg, distance_image):
-        """Test if a `~regions.PixRegion` overlaps with an exclusion mask.
-
-        If the regions is outside the exclusion mask, return 'False'
-        """
-        x, y = pixreg.center.x, pixreg.center.y
-        try:
-            val = distance_image.data[np.round(y).astype(int), np.round(x).astype(int)]
-        except IndexError:
-            return False
-        else:
-            return val < pixreg.radius
-
     @staticmethod
     def _compute_xy(pix_center, offset, angle):
-        """Compute x, y position for a given position angle and offset.
-
-        # TODO: replace by calculation using `astropy.coordinates`
-        """
-        dx = offset * np.sin(angle)
-        dy = offset * np.cos(angle)
+        """Compute x, y position for a given position angle and offset."""
+        dx = offset * np.cos(angle)
+        dy = offset * np.sin(angle)
         x = pix_center.x + dx
         y = pix_center.y + dy
         return PixCoord(x=x, y=y)
 
+    def _create_rotated_reg(self, curr_angle):
+        """ Compute a rotated region"""
+        _test_pos = self._compute_xy(self._pix_center, self._offset, curr_angle)
+
+        # The function copy.deepcopy does not work for these regions classes
+        dict = { _ : getattr(self._pix_region, _) for _ in self._pix_region._repr_params}
+        _test_reg = self._pix_region.__class__(_test_pos, **dict)
+
+        if hasattr(_test_reg, 'angle'):
```

Comment:
What happens if `hasattr(_test_reg, 'angle')` is False? e.g. `SkyPolygon` has no angle.

Reply:
The current class does not support SkyPolygon, as mentioned in the documentation. Should I raise an error early in the class in case of Polygon Use?

Reply:
Yes, an error should be raised if this doesn't work for some regions (and a test added to make sure that it does).
```
$ python -c 'import this' | grep Error
Errors should never pass silently.
```